instance/tasks.py

```diff
@@ -59,21 +59,19 @@ def spawn_appserver(instance_ref_id, mark_active_on_success=False, num_attempts=
         appserver_id = instance.spawn_appserver()
         if appserver_id:
             if mark_active_on_success:
-                # If the AppServer provisioned successfully, make it the active one:
-                # Note: if I call spawn_appserver() twice, and the second one provisions sooner, the first one may then
-                # finish and replace the second as the active server. We are not really worried about that for now.
-                instance.set_appserver_active(appserver_id)
+                # If the AppServer provisioned successfully, make it active.
+                appserver_make_active(appserver_id)
```

Comment:
@smarnach  I disagree..  Currently, we only use the `mark_active_on_success` flag when creating appservers for PRs, and it's useful for that.

For GUI actions, we have to explicitly activate the new appserver, and so will have to remember to deactivate the old one too, once this change goes through.

And anyone who activates an appserver from the shell will have to do the same, so..  Is it really that confusing?  If it is, then we should probably add a field to the instance indicating whether multiple VMs are even possible for this instance, and maintain the current "activating an appserver means deactivating the previously active appserver" behaviour.